### PR TITLE
Logging implementation on the microservices

### PR DESCRIPTION
**Pull request by ayushgautam**
Changes:
1. Log.java file: addded new class to create a filewriter and writeFIle functionality
2. Individual services:
- Initialize Log
- Call Write File method in Log class
- passed in extra username parameter from the MSClientAPI to the individual service method calls for tracking each user's actions in case of multiple sessions
3. Dockerfiles:
- Each individual service is given the Log.java file via dockerfile
4. stack-ms.yml:
- mapped directory in docker system to local system directory
5. ./ms_logs:
- Directory where log file would get generated, file write happens in append mode

### CODE REVIEW

File: ms/ms.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (alecyuverymuch) [inline ms/ms.iml]: This is an Intellij file? We should remove this

File: ms_logs/logs.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Mon Feb 21 02:24:44 UTC 2022 | User signed up:Alok | USERNAME:N/A
```
Comment (alecyuverymuch) [inline ms_logs/logs.txt]: If we are writing to the server, why do we have a log file here?

Comment (achat2) [inline ms_logs/logs.txt]: Please keep `ms_logs` in .gitignore

File: ms/ms.iml (hunk 1 above)
Comment (achat2) [inline ms/ms.iml]: This also should go in .gitignore

File: ms/Log.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+{
+
+    public FileWriter createLogWriter()
+    {
+        FileWriter fileWriter = null;
+        try
+        {
+            //creating the log file
+            File myObj = new File("/logs/logs.txt");
+            if (myObj.createNewFile()) {
```
Comment (achat2) [inline ms/Log.java:20]: `createNewFile` would need exception handling. https://docs.oracle.com/javase/7/docs/api/java/io/File.html#createNewFile()

File: ms/AuthServices.java [hunk 4]
```diff
@@ -40,15 +44,23 @@ public static void main(String args[])
...
+            log = new Log();
+            fileWriter = log.createLogWriter();
+            //disable logging if the logging initialization failed
+            if(fileWriter == null)
+                logEnable = false;
         } catch (Exception e) {
 
             System.out.println("AuthServices binding err: " + e.getMessage()); 
             e.printStackTrace();
+            if(logEnable)
```
Comment (achat2) [inline ms/AuthServices.java]: Would it be better to put `logEnable` in Log.java instead of here?

**Review by achat2 — CHANGES_REQUESTED** [review]
Some comments to give general structure. You can choose to address/ignore whichever you want. 😄

File: ms/Log.java (hunk 3 above)
Comment (ayushgautam) [reply ms/Log.java:20]: So there is already a try catch block around that. Can we talk more on this ?

File: ms/ms.iml (hunk 1 above)
Comment (ayushgautam) [reply ms/ms.iml]: Done

File: ms_logs/logs.txt (hunk 2 above)
Comment (ayushgautam) [reply ms_logs/logs.txt]: Removed

Comment (ayushgautam) [reply ms_logs/logs.txt]: Removed from the checkin

File: ms/ms.iml (hunk 1 above)
Comment (ayushgautam) [reply ms/ms.iml]: removed

File: ms/AuthServices.java (hunk 4 above)
Comment (ayushgautam) [reply ms/AuthServices.java]: I feel it makes sense for each microservice to have their own flag of log enable and disable. Decoupling each service's functionality. Open for discussion.

**Pull request merged**